p2p_wallet/AppCoordinator/AppCoordinator+AppEventHandlerDelegate.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  AppCoordinator+AppEventHandlerDelegate.swift
+//  p2p_wallet
+//
+//  Created by Chung Tran on 23/05/2022.
+//
+
+import AnalyticsManager
+import Foundation
+import SolanaSwift
+
+extension AppCoordinator: AppEventHandlerDelegate {
+    func didStartLoading() {
+        window.showLoadingIndicatorView()
+    }
+
+    func didStopLoading() {
+        window.hideLoadingIndicatorView()
+    }
+
+    func createWalletDidComplete() {
+        isRestoration = false
+        navigateToOnboarding()
+        analyticsManager.log(event: .setupOpen(fromPage: "create_wallet"))
+    }
+
+    func restoreWalletDidComplete() {
+        isRestoration = true
+        navigateToOnboarding()
+        analyticsManager.log(event: .setupOpen(fromPage: "recovery"))
+    }
+
+    func onboardingDidFinish(resolvedName: String?) {
+        self.resolvedName = resolvedName
+        let event: AnalyticsEvent = isRestoration ? .setupWelcomeBackOpen : .setupFinishOpen
+        analyticsManager.log(event: event)
+        navigateToOnboardingDone()
+    }
+
+    func userDidChangeAPIEndpoint(to _: APIEndPoint) {
+        showAuthenticationOnMainOnAppear = false
+        Task {
+            await reload()
+        }
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.3) { [weak self] in
```

Comment:
why is the timing 0.3 ? is it supposed to happen after `reload()` finished ?

Reply:
Yes. Maybe the migration code from old Root to new Coordinator cause this line. I'll fix it.
Thank you!